src/matcha_ml/state/remote_state_manager.py

```diff
@@ -213,61 +200,31 @@ def provision_remote_state(
         state_storage_template.build_template(config, template, destination, verbose)
 
         account_name, container_name, resource_group_name = template_runner.provision()
-        self._write_matcha_config(account_name, container_name, resource_group_name)
+        properties = [
+            MatchaConfigComponentProperty(name="account_name", value=account_name),
+            MatchaConfigComponentProperty(name="container_name", value=container_name),
+            MatchaConfigComponentProperty(
+                name="resource_group_name", value=resource_group_name
+            ),
+        ]
+        remote_state_bucket_component = MatchaConfigComponent(
+            name="remote_state_bucket", properties=properties
+        )
+        matcha_config = MatchaConfig(components=[remote_state_bucket_component])
+        MatchaConfigService.write_matcha_config(matcha_config=matcha_config)
```

Comment:
Can this be updated to use an "update config function" if the config file already exists?

Reply:
Just as this currently overwrites the file and therefore it will not retain the "stack type" that has been set.

Reply:
This is part of a different ticket which has now been merged, happy to update it as part of this one however